addons/mail/static/tests/discuss/typing/typing.test.js

```diff
@@ -347,3 +359,30 @@ test("chat: correspondent is typing in chat window", async () => {
     );
     await contains("[title='Demo is typing...']", { count: 0 });
 });
+
+test("switching to another channel triggers notify_typing to stop", async () => {
+    const pyEnv = await startServer();
+    const userId = pyEnv["res.users"].create({ name: "Demo" });
+    const partnerId = pyEnv["res.partner"].create({
+        im_status: "online",
+        name: "Demo",
+        user_ids: [userId],
+    });
+    const chatId = pyEnv["discuss.channel"].create({
+        channel_member_ids: [
+            Command.create({ partner_id: serverState.partnerId }),
+            Command.create({ partner_id: partnerId }),
+        ],
+        channel_type: "chat",
+    });
+    pyEnv["discuss.channel"].create({ name: "general" });
+    onRpcBefore("/discuss/channel/notify_typing", (args) =>
+        step(`notify_typing:${args.is_typing}`)
+    );
+    await start();
+    await openDiscuss(chatId);
+    await insertText(".o-mail-Composer-input", "a");
+    await assertSteps(["notify_typing:true"]);
+    await click(".o-mail-DiscussSidebar-item", { text: "general" });
+    await assertSteps(["notify_typing:false"]);
```

Comment:
As I said in the 17.0 PR: you must keep `advanceTime()`, otherwise the test takes too long to await the debounce. HOOT has a hard limit to 5 seconds, so if the test has a cold run and a few steps, it can easily reach this threshold and make runbot red.